api/src/main/java/javax/config/ConfigValue.java

```diff
@@ -192,8 +209,8 @@
     String getKey();
 
     /**
-     * Returns the actual key which led to successful resolution and corresponds to the resolved value. This applies
-     * only when {@link #withLookupChain(String...)} is used.
+     * Returns the actual key which led to successful resolution and corresponds to the resolved value.
+     * This is mostly useful when {@link #withLookupChain(String...)} or {@link #evaluateVariables(boolean)} is used.
      * Otherwise the resolved key should always be equal to the original key.
      * This method is provided for cases, when arameterized resolution is
```

Comment:
Typo: probably should be "parametrized".

Reply:
Just wondered whether we should remove getResolvedKey at all? 
It only gives you the value from the last 'successful' resolve. And if this happens in parallel rather quickly you might end up with glitches from another thread.
Of course this will only happen if the underlying config changes, otherwise it will be idempotent anyway.

Reply:
Indeed, would be clever to log it out in case of onChange - If only we would not just have ditched that ;)